Src/ILGPU/Runtime/MemoryBufferExtensions.cs

```diff
@@ -1070,4 +1070,89 @@ partial class MemoryBuffer3D<T>
 
         #endregion
     }
+
+    /// <summary>
+    /// Extension methods for the allocation of memory buffers.
+    /// </summary>
+    public static class MemoryBufferExtensions
+    {
+        /// <summary>
+        /// Allocates a 1D memory buffer with the given content on the associated
+        /// accelerator.
+        /// </summary>
+        /// <typeparam name="T">The element type.</typeparam>
+        /// <param name="accelerator">The current accelerator.</param>
+        /// <param name="data">The initial data array.</param>
+        /// <returns>The allocated memory buffer.</returns>
+        public static MemoryBuffer<T> Allocate<T>(
+            this Accelerator accelerator,
+            T[] data)
+            where T : unmanaged
+        {
+            if (data is null)
+                throw new ArgumentNullException(nameof(data));
+            if (data.Length < 1)
```

Comment:
IMHO Allocating 0 bytes should not be an error, as it avoid having to catter for the special case all over the user code. It definitely works with malloc(), how does CUDA react?

Reply:
That makes sense to me. However, the current `MemoryBuffer` implementation does [not support](https://github.com/m4rs-mt/ILGPU/blob/master/Src/ILGPU/Runtime/MemoryBuffer.cs#L36) 0-length allocations.

Reply:
It looks like [cuMemAlloc](https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__MEM.html#group__CUDA__MEM_1gb82d2a09844a58dd9e744dc31e8aa467) does not like 0 byte allocations:
`If bytesize is 0, cuMemAlloc() returns CUDA_ERROR_INVALID_VALUE.`

Reply:
A possible workaround would be that we always allocate at least one element. This also avoids OpenCL-related problems - since OpenCL doesn't seem to like 0-length allocations 😢